### PR TITLE
Change password feature + code folding shortcut

### PR DESCRIPTION
**Pull request by AuHau**
Hey there,

I really like Markdown-notes and I was searching for some projects to contribute, so I will be happy to contribute here! :-) I have quiet some ideas for improvements, so I will first make tickets for them and discuss them with you. 

So far I implemented two things:
- Code folding with shortcut 
- Change password
- Password reset

There are few things to be discussed and checked:
- can you please do code review for my changes? I am new to Django, so I might approach some things not in optimal way and generally I will be happy for any feedback on my code.
- I did small styling change in the forms (label aligned to right), is it OK?
- we have to agree on the keyboard shortcut for the code folding toggle

Thanks and take care!

### CODE REVIEW

**Comment by nicbou** [comment]
Amazing work Adam! Thanks for the help! I'll have a look at the pull request and get back to you shortly.

**Comment by AuHau** [comment]
Great :-)
I also added Password reset feature. There might be some styling issues, so please check if it looks OK.

File: authentication/forms.py [hunk 1]
```diff
@@ -7,6 +7,39 @@ class UserForm(forms.ModelForm):
...
+                raise ValidationError(_(u'The old password you have entered is not correct.'))
+
+            if not new_password1 == new_password2:
+                raise ValidationError(_(u'New passwords are not matching.'))
+
+        return cleaned_data
+
+    def save(self, commit=True):
+        instance = super(UserForm, self).save(False)
+        instance.set_password(self.cleaned_data['new_password1'])
```
Comment (nicbou) [inline authentication/forms.py]: For the sake of clarity, I would rename this variable to "user"

**Comment by nicbou** [comment]
Everything else is solid. Once you correct the shortcut, I will merge and deploy. Good work!

**Comment by AuHau** [comment]
Done! 
It is bit funny, because the work on the feature "changing password" was bit useless, since I started to work on the implementation of the ApiKeyAuthentication, it will have to be redone in API way :laughing:

**Pull request merged**